extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/ErrorInterceptor.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+/**
+ * Error interceptor must be used to intercept
+ * {@link io.quarkus.websockets.next.runtime.WebSocketEndpoint#doOnError(Throwable)}.
+ * The 'doOnError' method is called from within the class and using an endpoint wrapper wouldn't be sufficient.
+ */
+public sealed interface ErrorInterceptor permits ErrorCountingInterceptor, ErrorInterceptor.NoopErrorInterceptor {
```

Comment:
+1 for starting to use `sealed` and `permits` :)

Reply:
yeah, I thought about it because of https://github.com/quarkusio/quarkus/issues/41546

Reply:
though it's still public, but better restrict as much as I can